services/venice-router/src/main/java/com/linkedin/venice/router/api/VeniceDispatcher.java

```diff
@@ -411,4 +411,25 @@ public void run() {
       }
     }
   }
+
+  public boolean hasInFlightRequests() {
+    AggRouterHttpRequestStats stats = routerStats.getStatsByType(RequestType.SINGLE_GET);
```

Comment:
By default, Tehuti is using 30s as the sliding window, which means if there is one request in the first second, the metric value won't be 0 until 30s later, which can potentially delay the Router shutdown, and today, the window is fixed.
I wondered whether we should introduce a new metric to capture the inflight requests for all kind of requests and the time-window is configurable, so that we can better control the delay duration.
What do you think?

Reply:
We could introduce the new metric to track total in-flight for all type of requests with shorter period, but saving extra few seconds during router shutdown which are generally fast, at the cost of more frequent metric collection may not be worth it?

Reply:
Also total extra time added due to this change is limited to 10x10s. We may actually increase it bit to say 5x30s to account for metric collection delay.

Reply:
Short window is not more frequent metric collections, and it is just trying to use a short window and we don't need to report this metric. 
The meta point is that we are using the existing metrics in a different way, which is not optimized for the inflight requests check purpose, and the new metric will help that and we can make the time window configuragble, and it won't affect other existing metrics.

Today, in the steady state, each Router may only take 10 seconds to shutdown, and with the new logic, it can increase the deployment time (no idea how much), and if it does increase it a lot, we don't have a way to tune it.

Router layer is normally using a large fleet.